coriolis/tests/api/middleware/test_auth.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2022 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from unittest import mock
+
+from oslo_middleware import request_id
+from oslo_serialization import jsonutils
+
+from coriolis.api.middleware.auth import CoriolisKeystoneContext
+from coriolis.api import wsgi
+from coriolis import context
+from coriolis.tests import test_base
+
+
+class CoriolisKeystoneContextTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis api middleware auth."""
+
+    @mock.patch.object(jsonutils, "loads")
+    @mock.patch.object(context, "RequestContext")
+    def test__call__(self, mock_request_context, mock_jsonutils_loads):
+        req_mock = mock.Mock()
+        headers_mock = mock.Mock()
+        environ_mock = {}
+        remote_addr_mock = mock.Mock()
```

Comment:
You don't need to mock every attribute of `req_mock`. The code you're testing will set these for you. Only keep `req_mock`, and directly set `req_mock.headers = { key: value}`.